test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -383,8 +391,62 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Single(activityProcessor.Invocations, invo => invo.Method.Name == "OnEnd");
             var activity = activityProcessor.Invocations.FirstOrDefault(invo => invo.Method.Name == "OnEnd").Arguments[0] as Activity;
 
-            Assert.Equal(ActivityKind.Server, activity.Kind);
-            Assert.Equal("/api/values", activity.GetTagValue(SpanAttributeConstants.HttpPathKey) as string);
+            ValidateAspNetCoreActivity(activity, "/api/values");
+        }
+
+        [Fact]
+        public async Task PropagateContextIrrespectiveOfSamplingDecision()
+        {
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+
+            var expectedTraceId = ActivityTraceId.CreateRandom();
+            var expectedParentSpanId = ActivitySpanId.CreateRandom();
+
+            // Arrange
+            using (var testFactory = this.factory
+                .WithWebHostBuilder(builder =>
+                    builder.ConfigureTestServices(services =>
+                    {
+                        this.openTelemetrySdk = Sdk.CreateTracerProviderBuilder()
+                        .SetSampler(new AlwaysOffSampler())
+                        .AddAspNetCoreInstrumentation()
+                        .AddProcessor(activityProcessor.Object)
+                        .Build();
+                    })))
+            {
+                using var client = testFactory.CreateClient();
+
+                // Test TraceContext Propagation
+                var expectedTraceState = "rojo=1,congo=2";
+                var request = new HttpRequestMessage(HttpMethod.Get, "/api/GetChildActivityTraceContext");
+                request.Headers.Add("traceparent", $"00-{expectedTraceId}-{expectedParentSpanId}-01");
```

Comment:
portions of this test would pass by default, as asp.net creates activity from traceparent always, irrespective of sampling in Otel.
To ensure thorough test, use a custom propagator (with some header like Foo), and validate the scenarios. Also add cases where sending traceparent with "00" and "01" as sampling flags